operators/src/machine_learning/onnx.rs

```diff
@@ -54,11 +56,37 @@ impl RasterOperator for Onnx {
 
         let model_metadata = context.ml_model_metadata(&self.params.model).await?;
 
+        let tiling_shape = context.tiling_specification().tile_size_in_pixels;
+
+        // check that we can use the model input shape with the operator
+        ensure!(
```

Comment:
isn't this checked by try_dimensions_to_tensorshape?

Reply:
no this check is that the tensor is compatible to the tiling_shape the engine uses.

Reply:
i moved all of the testing to onnx_utils.rs we need to check the number of input bands in the operator because we need the ResultDescriptor of the specific workflow for that. (check_model_input_features)